ui/src/components/EditReference/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import * as Interfaces from '@interfaces';
+import * as Components from '@components';
+import * as tiptap from '@tiptap/react';
+import * as Config from '@config';
+
+import StarterKit from '@tiptap/starter-kit';
+import Underline from '@tiptap/extension-underline';
+interface MenuBarProps {
+    editor: tiptap.Editor;
+    loading: boolean;
+    setLoading: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+type Props = {
+    positiveActionCallback: (referenceID: string, referenceContent: string) => void;
+    negativeActionCallback: () => void;
+    MenuBar: React.FC<MenuBarProps>;
```

Comment:
MenuBar component needs to be extracted from the MainText in it's own folder/file and imported from there